doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+        container ROLE_TABLE {
+
+            description "ROLE MANAGEMENT part of config_db.json";
+
+            list ROLE_TABLE_LIST {
+
+                key "name";
+
+                leaf name {
+                  type string {
+                        pattern '[a-zA-Z_]([a-zA-Z0-9_-]{0,31}|[a-zA-Z0-9_-]{0,30}$)';
+                        length 1..32 {
+                              error-message "Invalid length for the role-name.";
+                              error-app-tag rolename-invalid-length;
+                        }
+                  }
+                  description "Role name.";
+                }       
+
+                leaf primary_group {
+                    mandatory true;
+                    type string;
+                  	description "Primary linux group.";                    
+                }
+                
+                leaf secondary_groups {
+                    mandatory true;
+                    type string;
+                  	description "List of secondary linux groups seperated by comma.";                    
+                }                
+
+               
+            } /* end of list ROLE_TABLE_LIST */
+
+        } /* end of container ROLE_TABLE */
+
+    } /* end of container sonic-role-mgmt */
+
+} /* end of module sonic-role-mgmt */
+
+
+```
+
+##### Config CLI
+
+```
+==============================================================================
+root@host:~$ config username add --help
+Usage: config user add [OPTIONS] <username> <password>
+
+  Add new user with password
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+==============================================================================
+root@host:~$ config username del --help
+Usage: config user del [OPTIONS] <username>
+
+  Delete a user
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+==============================================================================
+root@host:~$ config username state --help
+Usage: config user state [OPTIONS] <username> <state>
+
+  Specify a user state [enabled | disabled]
+
+Options:
+  -?, -h, --help  Show this message and exit.
+==============================================================================
+root@host:~$ config username full-name --help
+Usage: config user full-name [OPTIONS] <username> <full-name>
+
+  Specify a user full-name (Gecos field)
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+==============================================================================
+root@host:~$ config username role --help
+Usage: config user role [OPTIONS] <username> <role>
+
+  Specify a user role (capabilities)
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+==============================================================================
+root@host:~$ config username password --help
+Usage: config user password [OPTIONS] <username> <password>
+
+  Specify a user password
+
+Options:
+  -?, -h, --help  Show this message and exit.
+```
```

Comment:
Please consider providing configuration cli for new role additions as well.

Reply:
I will add the role commands to the HLD
but I think that we won't implement it them at phase 1 of this feature